index.sh

```diff
@@ -298,10 +309,11 @@ EOM
     else
       cat 1>&2 << 'EOM'
 /=====================================\\
-|       INSUFFICIENT PRIVILEGES       |
+|    PLEASE READ THIS ERROR MESSAGE    |
```

Comment:
Couple ideas here:
- Could we just avoid prompting the user if they don't have permission to write to `/etc/alternatives`? E.g. if they don't have write permission:
  -  if `GETMICRO_REGISTER` is undefined, print a brief warning letting them know that `update-alternatives` support is available but they would need to have root privileges (so the next time they use `getmicro`, they can use `sudo` and get this feature
  - If `GETMICRO_REGISTER=y`, abort with this error message
  - If `GETMICRO_REGISTER=n`, don't print any error or warning

Reply:
Another option: could we detect whether the system supports `sudo` and if so offer to just elevate permissions?

E.g. `Insufficient permissions to use update-alternatives. Try again with sudo? [y/N]: ` and then just re-run the block above with sudo if they select `y`.

Reply:
I think I have the solution: the error message gives you the option to abort trying to use update-alternatives.

We can't automatically run sudo because then people like mrx23dot will install micro into their home directory, which will cause all sorts of issues when another user tries to use a system tool dependent on editor and can't run editor due to lack of permissions.

Reply:
I love your solution to inform people about the GETMICRO_REGISTER but not prompt them by default.

This way, people who read will benefit from this feature and people who don't won't bother you, which is the best of both worlds.

The only issue with this is that it severely detracts from ease of use. The majority of people installing micro likely want this feature, so we shouldn't penalize them for this. 

Thus, I propose that we add a http://getmic.ro/r script. No extra units tests needed as this script is the same as index.sh except that it has an added `GETMICRO_REGISTER=y` at the start and is auto-generated.

Your other solution--to run sudo--would turn out quite problematic for people like mrx23dot. We can't automatically run sudo because people will use update-alternatives and install micro into their home directory, which will cause all sorts of issues when another user tries to use a system tool dependent on editor and can't run editor due to lack of permissions.

Reply:
I just realized another issue with prompting for sudo and why making this an opt-in no-prompts feature really is the best way to go: multi-lingual speakers who struggle with English.